cdap-gateway/src/main/java/co/cask/cdap/gateway/router/RouterPathLookup.java

```diff
@@ -88,6 +88,7 @@ private String getV3RoutingService(String [] uriParts, AllowedMethod requestMeth
       matches(uriParts, "v3", "namespaces", null, "streams", null, "metadata", "tags") ||
       matches(uriParts, "v3", "namespaces", null, "metadata", "search") ||
       matches(uriParts, "v3", "namespaces", null, "datasets", null, "lineage") ||
+      matches(uriParts, "v3", "namespaces", null, "streams", null, "lineage") ||
```

Comment:
add a verification for this in ``RouterPathTest.testMetadataRouting()`` as well.